src/modules/pythonmonkey/pythonmonkey.cc

```diff
@@ -210,57 +246,14 @@ PyTypeObject JSObjectItemsProxyType = {
 };
 
 static void cleanup() {
+  delete jsFunctionRegistry;
```

Comment:
this block below at line 375 seems outdated:

bool rvalIsFunction = cls == js::ESClass::Function; // function object
bool rvalIsString = rval.isString() || cls == js::ESClass::String; // string primitive or boxed String object
if (!(rvalIsFunction || rvalIsString)) { // rval may be a JS function or string which must be kept alive.
delete rval;
}

Reply:
done for functions, for strings still not handled but thats not within the scope of this MR

Reply:
let's please do it now that we are in-context